lib/_nonstd.scm

```diff
@@ -3098,9 +3098,7 @@
            (let* ((prefix
                    (or path
                        (##path-expand
-                        (##string-append
-                         (##path-strip-directory (##executable-path))
-                         "-temp")
+                        "gsx-temp"
```

Comment:
Using a fixed name, and in particular `gsx-temp`, is not a good idea because it will not be related to the program that created the directory.

A primitive procedure `##process-identifier-string` should be added that returns a string related to the program's name and computed in steps, returning the first that doesn't fail :

1) if `##processed-command-line` is a pair, then `(##path-strip-extension (##path-strip-directory (##car ##processed-command-line)))`
2) same for `(##os-executable-path)`
3) `"process"`

and then the pid is added as a suffix with `(##string-append ... "-" (##number->string (##os-getpid)))`.

Then `##create-temporary-directory` can be changed to avoid adding the pid.

Reply:
This makes sense and is definitely the smarter solution.  My thought process in choosing "gsx-temp" was essentially "the smaller the change, the more likely it'll be accepted."  Rookie mistake.

Reply:
No problem. By the way the name `##process-identifier-string` is a bit confusing because it could be interpreted as the process PID in string form. So maybe `##process-mnemonic-identifier` is better. Sufficiently unusual that it won't be confused with more common process functionality.